src/libraries/System.Runtime/tests/System/Runtime/CompilerServices/RuntimeHelpersTests.cs

```diff
@@ -193,16 +192,100 @@ private static void FillStack(int depth)
             }
         }
 
+        public static IEnumerable<object[]> GetUninitializedObject_NegativeTestCases()
+        {
+            // TODO: Test actual function pointer types when typeof(delegate*<...>) support is available
+
+            yield return new[] { typeof(string), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { typeof(int[,]), typeof(ArgumentException) }; // variable-length type
+            yield return new[] { Array.CreateInstance(typeof(int), new[] { 1 }, new [] { 1 }).GetType(), typeof(ArgumentException) }; // variable-length type (non-szarray)
+            yield return new[] { typeof(Array), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(Enum), typeof(MemberAccessException) }; // abstract type
+
+            yield return new[] { typeof(Stream), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(IDisposable), typeof(MemberAccessException) }; // interface type
+
+            yield return new[] { typeof(List<>), typeof(MemberAccessException) }; // open generic type
+            yield return new[] { typeof(List<>).GetGenericArguments()[0], typeof(ArgumentException) }; // 'T' placeholder typedesc
+
+            yield return new[] { typeof(Delegate), typeof(MemberAccessException) }; // abstract type
+            yield return new[] { typeof(Action), typeof(ArgumentException) }; // delegate type
+
+            yield return new[] { typeof(void), typeof(ArgumentException) }; // explicit block in place
+            yield return new[] { typeof(int).MakePointerType(), typeof(ArgumentException) }; // pointer typedesc
+            yield return new[] { typeof(int).MakeByRefType(), typeof(ArgumentException) }; // byref typedesc
+
+            yield return new[] { typeof(ReadOnlySpan<int>), typeof(NotSupportedException) }; // byref type
+            yield return new[] { typeof(ArgIterator), typeof(NotSupportedException) }; // byref type
+
+            if (PlatformDetection.IsNetCore)
+            {
+                Type canonType = typeof(object).Assembly.GetType("System.__Canon", throwOnError: true);
```

Comment:
There was a discussion about this specific test in the past: https://github.com/dotnet/runtime/pull/36604#discussion_r426199119

Reply:
Isn't it good though to get this low level test coverage? If we have code paths, its good to test them.

Reply:
There are plenty of ways to destabilize the system via private reflection. If we were to delete theses checks from the runtime like Jan suggested in the conversation I linked to, nothing horrible would happen. Adding tests for this implies we care about this behavior which then begs the question why this is not testing the implementation details on Mono as well (Mono doesn't use `__Canon`) and if Mono doesn't have such checks, why it doesn't have them. It's a lot of effort for nothing.

If I want to get a magic `__Canon` instance that e.g. defeats type casting, I might as well private-reflect on `System.Runtime.CompilerServices.CastHelpers` and adjust casting behavior the way I want. We make no guarantees once somebody private-reflects.